src/kakarot/state.cairo

```diff
@@ -125,6 +126,114 @@ namespace State {
         }
     }
 
+    func cache_precompiles{
+        syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, state: model.State*
+    }() {
+        alloc_locals;
+        tempvar accounts_ptr = state.accounts;
+        with accounts_ptr {
+            Internals._cache_precompile(1);
+            Internals._cache_precompile(2);
+            Internals._cache_precompile(3);
+            Internals._cache_precompile(4);
+            Internals._cache_precompile(5);
+            Internals._cache_precompile(6);
+            Internals._cache_precompile(7);
+            Internals._cache_precompile(8);
+            Internals._cache_precompile(9);
+        }
+        tempvar state = new model.State(
+            accounts_start=state.accounts_start,
+            accounts=accounts_ptr,
+            events_len=state.events_len,
+            events=state.events,
+            transfers_len=state.transfers_len,
+            transfers=state.transfers,
+        );
+
+        return ();
+    }
+
+    func cache_access_list{
+        syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, state: model.State*
+    }(access_list_len: felt, access_list: model.AccessListItem*) -> felt {
+        alloc_locals;
+        tempvar accounts_ptr = state.accounts;
+        with accounts_ptr {
+            let gas_cost = _cache_access_list(access_list_len, access_list);
+        }
+        tempvar state = new model.State(
+            accounts_start=state.accounts_start,
+            accounts=accounts_ptr,
+            events_len=state.events_len,
+            events=state.events,
+            transfers_len=state.transfers_len,
+            transfers=state.transfers,
+        );
+
+        return gas_cost;
+    }
+
+    func _cache_access_list{
+        syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, accounts_ptr: DictAccess*
+    }(access_list_len: felt, access_list: model.AccessListItem*) -> felt {
+        alloc_locals;
+        if (access_list_len == 0) {
+            return 0;
+        }
+
+        let access_list_item = [access_list];
+        let address = access_list_item.address;
+        let account = Account.fetch_or_create(address);
+        tempvar storage_ptr = account.storage;
+        with storage_ptr {
+            Account.cache_storage_keys(
+                access_list_item.storage_keys_len, access_list_item.storage_keys
+            );
+
+            tempvar account = new model.Account(
+                address=account.address,
+                code_len=account.code_len,
+                code=account.code,
+                storage_start=account.storage_start,
+                storage=storage_ptr,
+                nonce=account.nonce,
+                balance=account.balance,
+                selfdestruct=account.selfdestruct,
+            );
+        }
```

Comment:
close the bracket right after `Account.cache_storage_keys` as it's not used as implicit for `new model.Account`